eng/Version.Details.xml

```diff
@@ -98,7 +91,6 @@
     <Dependency Name="Microsoft.Net.Compilers.Toolset" Version="4.11.0-3.24378.3">
       <Uri>https://github.com/dotnet/roslyn</Uri>
       <Sha>5e3a11e2e7f952da93f9d35bd63a2fa181c0608b</Sha>
-      <SourceBuild RepoName="roslyn" ManagedOnly="true" />
```

Comment:
Why was this removed?

Reply:
After new arkade sdk was consumed, the source build was failing on this one, with errors: https://dev.azure.com/dnceng-public/public/_build/results?buildId=762782&view=logs&j=c316b76e-5b95-5e79-2740-62e92f13eebd&t=e33df8ad-5872-5105-0672-039423de0ed3 Only place <SourceBuild> elements on source-build intermediate nupkgs.

Reply:
The correct fix is to add the corresponding soucebuild intermediate dependency (ideally right after this dependency)

```
    <Dependency Name="Microsoft.SourceBuild.Intermediate.roslyn" Version="4.11.0-3.24378.3">
      <Uri>https://github.com/dotnet/roslyn</Uri>
      <Sha>5e3a11e2e7f952da93f9d35bd63a2fa181c0608b</Sha>
      <SourceBuild RepoName="roslyn" ManagedOnly="true" />
    </Dependency>
```

Then delete the change to https://github.com/dotnet/msbuild/pull/10451/files#diff-f88ce4026366ce74fb1a981785c542757617abf4393a33ca34465a5e9a0dd343